### DISCUSSION TITLE
How to set vim.g.rustaceanvim before rustaceanvim is initialized?

### DISCUSSION THREAD

**Discussion by WhiteBlackGoose** [mrcjkb/rustaceanvim#636] [Q&A]
### Checked other resources

- [X] This is not a bug report or feature request (Please use [issues](https://github.com/mrcjkb/rustaceanvim/issues/new/choose) for those).
- [X] This is a question about rustaceanvim, not rust-analyzer or Neovim.
- [X] This is [not a question about autocompletion](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#im-having-issues-with-autocompletion).
- [X] I have read the README.
- [X] I have read the [FAQ](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#left_speech_bubble-faq).
- [X] I have searched for a solution using `:h rustaceanvim`.
- [X] I have searched [existing discussions](https://github.com/mrcjkb/rustaceanvim/discussions).
- [X] I have searched the existing issues/discussions of other plugins related to this question (if applicable).
- [X] I have searched the [existing rust-analyzer discussions](https://github.com/rust-lang/rust-analyzer/discussions) (if applicable).
- [X] I have added a very descriptive title to this question.

### Description

Hi there again,

rustaceanvim doesn't have `setup{}` and it's unclear how to ensure `vim.g.rustaceanvim` is read before rustaceanvim is initialized, at least I didn't find a way.

`:checkhealth` returns
```
vim.g.rustaceanvim is set, but it was sourced after rustaceanvim was initialized
```

When does `rustaceanvim` initialize? It even used to work, but sadly now I can't make it work again. Thanks in advance!

### Example Code

```lua
.
```

### Neovim version (nvim -v)

.

### Operating system/version

.

### Output of :checkhealth rustaceanvim

```console
P. S. Please allow creating normal discussions and not just this template
```

**Comment by mrcjkb** [comment]
Hey :wave: 

> P. S. Please allow creating normal discussions and not just this template

This template exists for a good reason: 
I put my free time into this plugin. The template encourages and guides people on how to do some research/troubleshooting before opening a new discussion.

For example, a search for the `checkhealth` error message yields a related discussion that may have the answer to your question:

- https://github.com/mrcjkb/rustaceanvim/discussions/77

Rustaceanvim uses `:h ftplugin` to initialise.

**Reply by WhiteBlackGoose** [reply]
Hi, 
> The template encourages and guides people on how to do some research/troubleshooting before opening a new discussion.

The checkboxes are fine, I meant the other blocks.

> For example, a search for the checkhealth error message yields a related discussion that may have the answer to your question

Sadly not, because I don't use after/** stuff (at least from my understanding that would be the problem?).

However there appears to be some sort of a race condition. I sadly am not knowledgeable enough in plugin craftsmanship, but basically I moved `rustaceanvim` a bit up in the initialization and the problem was gone. I moved it back and ... the problem did *not* reappear. So now it works, even though the config is intact. I have a fair amount of plugins in my config, and it's sadly not something that can be reproduced with a minimal config... so I'll keep watching out and try to find a repro.

Thanks :ok_hand:

**Reply by mrcjkb** [reply]
That indicates that it's likely caused by [this](https://github.com/mrcjkb/rustaceanvim/discussions/77#discussioncomment-7978623).

**Reply by WhiteBlackGoose** [reply]
hitting this bug again... is there a way to call some equivalent of setup{} to force it to reinitialize?
```
Checking config ~
- ERROR Failed to run healthcheck for "rustaceanvim" plugin. Exception:
  ...share/nvim/lazy/rustaceanvim/lua/rustaceanvim/health.lua:127: vim.g.rustaceanvim is set, but it was sourced after rustaceanvim was initialized.
  

```

**Reply by mrcjkb** [reply]
There's no supported way to do this, but a hacky way to force evaluation would be to  `require("rustaceanvim.config.internal")`.

I don't recommend this - instead I recommend finding out why your ftplugins are being sourced prematurely, as this bug could lead to other subtle issues.
